dotnet/csharp.bzl

```diff
@@ -431,48 +440,88 @@ def _nuget_package_impl(repository_ctx):
   # Lastly we add the nuget package name.
   nuget_cmd += [repository_ctx.attr.package]
   # execute nuget download.
-  repository_ctx.execute(nuget_cmd)
-  # TODO(jeremy): report errors if there were any
-
-  tpl_file = Label("//dotnet:NUGET_BUILD.tpl")
-  # add the BUILD file
-  repository_ctx.template(
-    "BUILD",
-    tpl_file,
-    {"%{package}": repository_ctx.name,
-     "%{output_dir}": "%s" % output_dir})
+  result = repository_ctx.execute(nuget_cmd)
+  if result.return_code:
+    fail("Nuget command failed: %s (%s)" % (result.stderr, " ".join(nuget_cmd)))
+
+  if build_file_content:
+    repository_ctx.file("BUILD", build_file_content)
+  elif build_file:
+    repository_ctx.symlink(repository_ctx.path(build_file), "BUILD")
+  else:
+    tpl_file = Label("//dotnet:NUGET_BUILD.tpl")
+    # add the BUILD file
+    repository_ctx.template(
+      "BUILD",
+      tpl_file,
+      {"%{package}": repository_ctx.name,
+       "%{output_dir}": "%s" % output_dir})
+
+_nuget_package_attrs = {
+  # Sources to download the nuget packages from
+  "package_sources":attr.string_list(),
+  # The name of the nuget package
+  "package":attr.string(mandatory=True),
+  # The version of the nuget package
+  "version":attr.string(mandatory=True),
+  # Reference to the mono binary
+  "mono_exe":attr.label(
+    executable=True,
+    default=Label("@mono//bin:mono"),
+  ),
+  # Reference to the nuget.exe file
+  "nuget_exe":attr.label(
+    default=Label("@nuget//:nuget.exe"),
+  ),
+}
 
 # This rule is a repository rule and is only usable in WORKSPACE files.
 # due to some limitations of repository_rules it does require you to
```

Comment:
We no longer require them to say where the nuget binary is now right? We should remove that from this comment.